docker-compose-dev.yml

```diff
@@ -70,6 +70,46 @@ services:
     ports:
       - "8086:8086"
 
+  keycloak:
+    image: jboss/keycloak:8.0.2
+    hostname: keycloak
+    environment:
+      KEYCLOAK_USER: admin
+      KEYCLOAK_PASSWORD: adminadmin
+      DB_VENDOR: h2
+      # Can't properly provision a login theme
+#    volumes:
+#      - ${PWD}/xain-theme-keycloak:/opt/jboss/keycloak/themes
+#      - ${PWD}/xain-theme-keycloak/standalone.xml:/opt/jboss/keycloak/standalone/configuration/standalone.xml
+    networks:
+      - xain-fl-dev
+    ports:
+      - "8080:8080"
+
+  grafana:
+    image: grafana/grafana
+    depends_on:
+      - keycloak
+    hostname: grafana
+    ports:
+      - "3000:3000"
+    networks:
+      - xain-fl-dev
+    environment:
+      GF_AUTH_GENERIC_OAUTH_ENABLED: "true"
+      GF_AUTH_GENERIC_OAUTH_NAME: "OAuth"
+      GF_AUTH_GENERIC_OAUTH_CLIENT_ID: "grafana"
+      GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET: "c3fbd14d-dfe9-4dc3-9894-8a97bc6c6575"
+      GF_AUTH_GENERIC_OAUTH_ALLOW_SIGN_UP: "true"
+      GF_AUTH_GENERIC_OAUTH_SCOPES: "openid profile email"
+      #GF_AUTH_GENERIC_OAUTH_EMAIL_ATTRIBUTE_NAME: "email:primary"
+      GF_AUTH_GENERIC_OAUTH_AUTH_URL: "http://localhost:8080/auth/realms/master/protocol/openid-connect/auth"
```

Comment:
This should be `keycloak` instead of `localhost`, right?

Reply:
Nope, this is correct. This is the domain that the browser gets redirected to